extension/httpfs/create_secret_functions.cpp

```diff
@@ -131,4 +132,84 @@ void CreateS3SecretFunctions::RegisterCreateSecretFunction(DatabaseInstance &ins
+	secret_type.default_provider = "config";
+	ExtensionUtil::RegisterSecretType(instance, secret_type);
+
+	// Generic Bearer config provider
+	CreateSecretFunction config_fun = {GENERIC_BEARER_TYPE, "config", CreateBearerSecretFromConfig};
+	config_fun.named_parameters["token"] = LogicalType::VARCHAR;
+	ExtensionUtil::RegisterFunction(instance, config_fun);
+
+	// HuggingFace secret
+	SecretType secret_type_hf;
+	secret_type_hf.name = HUGGINGFACE_TYPE;
+	secret_type_hf.deserializer = KeyValueSecret::Deserialize<KeyValueSecret>;
+	secret_type_hf.default_provider = "config";
+	ExtensionUtil::RegisterSecretType(instance, secret_type_hf);
+
+	// Huggingface config provider
+	CreateSecretFunction hf_config_fun = {HUGGINGFACE_TYPE, "config", CreateBearerSecretFromConfig};
+	hf_config_fun.named_parameters["token"] = LogicalType::VARCHAR;
+	ExtensionUtil::RegisterFunction(instance, hf_config_fun);
+
+	// Huggingface cache provider
+	CreateSecretFunction hf_cache_fun = {HUGGINGFACE_TYPE, "cache", CreateHuggingFaceSecretFromCache};
+	ExtensionUtil::RegisterFunction(instance, hf_cache_fun);
+}
+
+unique_ptr<BaseSecret> CreateBearerTokenFunctions::CreateSecretFunctionInternal(ClientContext &context,
+                                                                                CreateSecretInput &input,
+                                                                                const string &token) {
+	// Set scope to user provided scope or the default
+	auto scope = input.scope;
+	if (scope.empty()) {
+		if (input.type == GENERIC_BEARER_TYPE) {
+			scope.push_back("");
+		} else if (input.type == HUGGINGFACE_TYPE) {
+			scope.push_back("hf://");
+		} else {
+			throw InternalException("Unknown secret type found in httpfs extension: '%s'", input.type);
+		}
+	}
+	auto return_value = make_uniq<KeyValueSecret>(scope, input.type, input.provider, input.name);
+
+	//! Set key value map
+	return_value->secret_map["token"] = token;
+
+	//! Set redact keys
+	return_value->redact_keys = {"token"};
+
+	return std::move(return_value);
+}
+
+unique_ptr<BaseSecret> CreateBearerTokenFunctions::CreateBearerSecretFromConfig(ClientContext &context,
+                                                                                CreateSecretInput &input) {
+	string token;
+
+	auto token_input = input.options.find("token");
+	for (const auto &named_param : input.options) {
+		auto lower_name = StringUtil::Lower(named_param.first);
+		if (lower_name == "token") {
+			token = named_param.second.ToString();
+		}
+	}
+
+	return CreateSecretFunctionInternal(context, input, token);
+}
+
+unique_ptr<BaseSecret> CreateBearerTokenFunctions::CreateHuggingFaceSecretFromCache(ClientContext &context,
+                                                                                    CreateSecretInput &input) {
+	LocalFileSystem fs;
+
+	auto handle = fs.OpenFile("~/.cache/huggingface/token", {FileOpenFlags::FILE_FLAGS_READ});
```

Comment:
Thanks for the ping!

Just wanted to add some context about how to retrieve the user token. The logic in `huggingface_hub` is:
- either check the [`HF_TOKEN`](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/environment_variables#hfhome) environment variable => directly read token from env
- either check [`HF_TOKEN_PATH`](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/environment_variables#hftokenpath) environment variable (can overwrite default path)
- either check [`$HF_HOME/token`](https://huggingface.co/docs/huggingface_hub/main/en/package_reference/environment_variables#hftoken) => defaults to `~/.cache/huggingface/token` if environment variable is not set.

Reply:
Thanks for clarifying this @Wauplin. I will add a `credential_chain` provider then instead of the current `cache` provider as that seems more in line with other secret providers in duckdb.

This way we a user can do:
```SQL
CREATE SECRET hf1 (TYPE HUGGINGFACE, provider credential_chain);
```
which will perform the steps you described in that order, creating the secret once one of the methods succeeds to fetch a token